plugins/connection/saltstack.py

```diff
@@ -85,9 +86,9 @@ def put_file(self, in_path, out_path):
 
         out_path = self._normalize_path(out_path, '/')
         self._display.vvv("PUT %s TO %s" % (in_path, out_path), host=self.host)
-        with open(in_path) as in_fh:
+        with open(in_path, 'rb') as in_fh:
             content = in_fh.read()
-        self.client.cmd(self.host, 'file.write', [out_path, content])
+        self.client.cmd(self.host, 'hashutil.base64_decodefile', [codecs.encode(content, 'base64'), out_path])
```

Comment:
Could you please add a change log entry for this change - [here](https://github.com/ansible-collections/community.general/tree/main/changelogs/fragments)?

Reply:
Yes, added one just now.